cmd/reactGen/gen.go

```diff
@@ -0,0 +1,308 @@
+		}
+
+		for fn, file := range pkg.Files {
+
+			if gogenerate.FileGeneratedBy(fn, "reactGen") {
+				continue
+			}
+
+			foundImp := false
+			impName := ""
+
+			for _, i := range file.Imports {
+				p := strings.Trim(i.Path.Value, "\"")
+
+				if p == ReactPkg {
+					foundImp = true
+
+					if i.Name != nil {
+						impName = i.Name.Name
+					} else {
+						impName = path.Base(ReactPkg)
+					}
+
+					break
+				}
+			}
+
+			if !foundImp {
+				continue
+			}
+
+			for _, d := range file.Decls {
+				switch d := d.(type) {
+				case *ast.FuncDecl:
+					if d.Recv == nil {
+						continue
+					}
+
+					f := d.Recv.List[0]
+
+					se, ok := f.Type.(*ast.StarExpr)
+					if !ok {
+						continue
+					}
+
+					id, ok := se.X.(*ast.Ident)
+					if !ok {
+						continue
+					}
+
+					g.methods[id.Name] = append(g.methods[id.Name], d)
+
+				case *ast.GenDecl:
+					if d.Tok != token.TYPE {
+						continue
+					}
+
+					for _, ts := range d.Specs {
+						ts := ts.(*ast.TypeSpec)
+
+						st, ok := ts.Type.(*ast.StructType)
+						if !ok || st.Fields == nil {
+							continue
+						}
+
+						foundAnon := false
+
+						for _, f := range st.Fields.List {
+							if f.Names != nil {
+								// it must be anonymous
+								continue
+							}
+
+							se, ok := f.Type.(*ast.SelectorExpr)
+							if !ok {
+								continue
+							}
+
+							if se.Sel.Name != CompDefName {
+								continue
+							}
+
+							id, ok := se.X.(*ast.Ident)
+							if !ok {
+								continue
+							}
+
+							if id.Name != impName {
+								continue
+							}
+
+							foundAnon = true
+						}
+
+						if foundAnon && strings.HasSuffix(ts.Name.Name, CompDefSuffix) {
+							g.components[ts.Name.Name] = ts
+						} else {
+							g.types[ts.Name.Name] = ts
+						}
+					}
+				}
+			}
+		}
+
+		// at this point we have the components and their methods
+		for cd := range g.components {
+			g.genComp(cd)
+		}
+	}
+}
+
+type compGen struct {
+	*gen
+
+	Name string
+
+	HasState        bool
+	HasProps        bool
+	HasGetInitState bool
+
+	buf *bytes.Buffer
+}
+
+func (g *gen) genComp(defName string) {
+
+	name := strings.TrimSuffix(defName, CompDefSuffix)
+
+	cg := &compGen{
+		gen:  g,
+		buf:  bytes.NewBuffer(nil),
+		Name: name,
+	}
+
+	_, hasState := g.types[name+StateTypeSuffix]
+	_, hasProps := g.types[name+PropsTypeSuffix]
+
+	cg.HasState = hasState
+	cg.HasProps = hasProps
+
+	if hasState {
+		for _, m := range g.methods[defName] {
+			if m.Name.Name != GetInitialState {
+				continue
+			}
+
+			if m.Type.Params != nil && len(m.Type.Params.List) > 0 {
+				continue
+			}
+
+			if m.Type.Results != nil && len(m.Type.Results.List) != 1 {
+				continue
+			}
+
+			rp := m.Type.Results.List[0]
+
+			id, ok := rp.Type.(*ast.Ident)
+			if !ok {
+				continue
+			}
+
+			if id.Name == name+StateTypeSuffix {
+				cg.HasGetInitState = true
+				break
+			}
+
+		}
+	}
+
+	cg.pf("// Do not edit: file generated by %v\n", reactGenCmd)
```

Comment:
Suggestion, consider using:

```Go
cg.pf("// Code generated by %v; DO NOT EDIT.\n", reactGenCmd)
```

That would match the currently proposed format in https://github.com/golang/go/issues/13560#issuecomment-277804473, which might end up being the final accepted one.